### ISSUE TITLE
TestRemoveForce flaky test?

### ISSUE THREAD

**Issue by elboulangero** [docker/cli#2882]
### Describe the bug

While running unit tests:

```
=== Failed
=== FAIL: cli/command/container TestRemoveForce/with_force (0.00s)
    rm_test.go:44: assertion failed: 
        --- removed
        +++ →
          []string{
          	"mycontainer",
        + 	"nosuchcontainer",
          }
        
    --- FAIL: TestRemoveForce/with_force (0.00s)

=== FAIL: cli/command/container TestRemoveForce (0.00s)
```

Context: building the package on the Debian infrastructure.

The issue happened on the `s390x` architecture, for the version `20.10.0`. Cf: https://buildd.debian.org/status/fetch.php?pkg=docker.io&arch=s390x&ver=20.10.0%2Bdfsg1-1&stamp=1607657904

During previous builds (version `20.10.0-rc1`), there was no issue.

I've seen this failed on the `mips` architecture as well, although I have no logs. And the last time I tried, the test passed.

All of that leads me to believe that the test might be flaky. However the code looks quite straightforward, so I'm a bit puzzled.

For background, this test was introduced in 9a071a993a7d15cc490937104b3d37fd6a339bc2

**Comment by elboulangero** [comment]
I just had this test failure again, this time on `amd64` architecure: https://buildd.debian.org/status/fetch.php?pkg=docker.io&arch=amd64&ver=20.10.2%2Bdfsg1-1&stamp=1610074719

This time it's the `without_force` test which fails:

```
=== Failed
=== FAIL: cli/command/container TestRemoveForce/without_force (0.00s)
Error: Error: No such container: nosuchcontainer
    rm_test.go:37: assertion failed: 
        --- removed
        +++ →
          []string{
        + 	"mycontainer",
          	"nosuchcontainer",
          }
        
    --- FAIL: TestRemoveForce/without_force (0.00s)

=== FAIL: cli/command/container TestRemoveForce (0.00s)
```

**Comment by thaJeztah** [comment]
Are tests ran with `-parallel`? The `removed` variable is global to both subtests; if they run in parallel, that would possibly explain this 🤔

**Comment by elboulangero** [comment]
> Are tests ran with -parallel?

No, we run these tests with the Makefile provided, ie. `make test-unit`. In the logs, we can see that it translates to `gotestsum  -- ${TESTDIRS:-github.com/docker/cli/cli [...] github.com/docker/cli/templates}`. No `-parallel`.

Also after checking the go documentation, it seems that a test must declare itself as parallelizable with `t.Parallel()` to run in parallel? It does not seem to be the case here.

**Comment by Benehiko** [comment]
This is no longer the case, closing this issue.
